include/swift/Driver/Job.h

```diff
@@ -296,6 +308,7 @@ class Job {
 
   const char *getExecutable() const { return Executable; }
   const llvm::opt::ArgStringList &getArguments() const { return Arguments; }
+  const char * const &getResponseFileArg() const { return ResponseFileArg; }
```

Comment:
Returning a pointer wouldn't actually break any of that, but I guess using a const reference for a value that won't change signifies that the lifetime is tied to something else a little more strongly?

Reply:
We did try just returning a `const char *`, but doing so would cause it to read junk memory. Myself, @allevato, and @harlanhaskins looked at this issue, and this is what we came up with that worked.

Reply:
`TaskQueue::addTask` takes an ArrayRef<char *>. If we return the value of `getResponseFileArg()` as a pointer, then the returned pointer will get promoted to an `ArrayRef` through `ArrayRef::ArrayRef(T &OneElt)`, and the reference passed in is a temporary stack slot storing the returned address, not the actual address inside the `Job` struct.

Reply:
That's horrible; nice work. Can you add that as a comment so that a later refactorer like me doesn't try to change it?

Reply:
Alternately I suppose you could return an ArrayRef here even though the backing store is the one `const char *`.

Reply:
Using the `ArrayRef` does appear to work actually; I went ahead and used that. Do you think it still needs a comment? The code is more straightforward now, I think.

Reply:
No, I think this version is okay. Though I personally would have kept using the accessor methods in the other function.